HotelBase/ReservationOperationsProvider/ReservationOperationsProviderSeasonPrice.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using HotelBooking.BookingExternalService;
+using HotelBooking.HotelExamples;
+using HotelBooking.Logger;
+using HotelBooking.PaymentExternalService;
+
+namespace HotelBooking.ReservationOperationsProvider
+{
+    class ReservationOperationsProviderSeasonPrice : ReservationOperationsProviderCommonFunctions,
+        IReservationOperationsProvider
+    {
+        internal readonly IBookingService BookingService;
+        internal readonly IPaymentService PaymentService;
+        internal readonly ILogger Logger;
+        private const double SeasonPrice = 250;
+        private const double PriceAfterSeason = 200;
+        private const string Id = "01";
+
+        public ReservationOperationsProviderSeasonPrice(IBookingService bookingService, 
+            IPaymentService paymentService,
+            ILogger logger)
+        {
+            BookingService = bookingService;
+            PaymentService = paymentService;
+            Logger = logger;
+        }
+
+        public KeyValuePair<string, bool> ProcessOperation(DateTime date,
+            double price,
+            int creditCardNumber,
+            string email,
+            HotelOperation operation)
+        {
+            bool stepSuccess = false;
+            string operationDescription = "";
+            switch (operation.Operation)
+            {
+                case Operation.BookRoom:
+                    stepSuccess = BookRoom(date);
+                    operationDescription = OperationDescriptions.Booking;
+                    break;
+                case Operation.CheckPrice:
+                    stepSuccess = CheckPrice(price, date);
+                    operationDescription = OperationDescriptions.PriceValidation;
+                    break;
+                case Operation.MakePayment:
+                    stepSuccess = MakePayment(creditCardNumber, price);
+                    operationDescription = OperationDescriptions.Payment;
+                    break;
+                case Operation.SendEmail:
```

Comment:
Why you have to repeated so many operations?
Lets discuss about this by the desk. 

Tomorrow is my last day before short - a week long holiday.

Reply:
corrected